### PR TITLE
{FIX} Bug fixes

### PR DESCRIPTION
**Pull request by wangzelin007**
1. Modify the replace logic and add some descriptions
2. Modify the OPENAI_DEFAULT_MSG environment variable and default_msg variable, replacing single quotes with double quotes.
3. Service error status code changed from 400 to 500
4. Only put the error type in the response and remove the error details.
5. Add new error type: The OpenAI API rate limit is exceeded.
image
image

### CODE REVIEW

File: API/common/service_impl/chatgpt.py [hunk 1]
```diff
@@ -33,23 +33,30 @@ def gpt_generate(user_msg: str, history_msg: List[Dict[str, str]]) -> Dict[str,
...
     if "out of my scope" in content and content.startswith("Sorry"):
         return None
     try:
-        content = json.loads(content.replace("\"", "\\\"").replace("'", '"'))
+        # If only single quotes exist, replace them with double quotes.
+        if "'" in content and not '"' in content:
+            content = json.loads(content.replace("'", '"'))
+        # In other cases, convert directly to json
+        else:
+            content = json.loads(content)
```
Comment (ReaNAiveD) [inline API/common/service_impl/chatgpt.py:49]: Can the parsing logic handle cases where the argument value is a JSON object?
For example, 
```json
{
    "command": "az stream-analytics input create",
    "arguments": [
        "--name",
        "--resource-group",
        "--job-name",
        "--type",
        "--serialization",
        "--datasource"
    ],
    "reason": "Create an input for the Stream Analytics job",
    "example": "az stream-analytics input create --name MyInput --resource-group MyResourceGroup --job-name MyJob --type Stream --serialization '{\"type\": \"Json\",\"properties\": {\"encoding\": \"UTF8\"}}' --datasource '{\"type\": \"Microsoft.Storage/Blob\",\"properties\": {\"storageAccounts\": [{\"accountName\": \"MyAccount\",\"accountKey\": \"MyKey\"}],\"container\": \"MyContainer\",\"pathPattern\": \"telemetry/{date}/{time}\"}}'",
    "result": [],
    "example_result": []
},
```

**Comment by wangzelin007** [comment]
GPT now returns correct responses even if they contain complex formatting such as:
```
{"scenario": "Create and use Azure Policy to enforce tagging for Azure resources", "commandSet": [{"command": "az policy definition create", "arguments": ["--name", "--rules", "--description", "--display-name"], "reason": "Create a new policy definition to enforce tagging", "example": "az policy definition create --name enforce-tagging --rules '{\"if\": {\"not\": {\"field\": \"tags\", \"exists\": \"true\"}}, \"then\": {\"effect\": \"deny\"}}' --description 'Enforce tagging on resources' --display-name 'Enforce Tagging'"}, {"command": "az policy assignment create", "arguments": ["--name", "--policy", "--scope"], "reason": "Assign the policy to a specific scope (e.g., subscription, resource group, etc.)", "example": "az policy assignment create --name enforce-tagging-assignment --policy enforce-tagging --scope /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}"}], "description": "Create a new Azure Policy to enforce tagging on Azure resources and assign it to a specific scope. This will deny the creation or update of resources without tags."}
```

File: API/common/service_impl/chatgpt.py [hunk 2]
```diff
@@ -33,23 +33,30 @@ def gpt_generate(user_msg: str, history_msg: List[Dict[str, str]]) -> Dict[str,
...
         return content
     except JSONDecodeError as e:
-        raise GPTInvalidResultException(content) from e
+        raise GPTInvalidResultException from e
```
Comment (zhoxing-ms) [inline API/common/service_impl/chatgpt.py:52]: Could we use log to record the detailed information for future troubleshooting?

Comment (zhoxing-ms) [inline API/common/service_impl/chatgpt.py]: Could we use log to record the detailed information for future troubleshooting?

**Review by ReaNAiveD — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

File: API/common/service_impl/knowledge.py [hunk 3]
```diff
@@ -181,5 +183,6 @@ def pass_verification(question, result):
         content = response["choices"][0]["message"]["content"]
         content = content.replace("\"", "").replace("'", "").lower()
         if content not in ['true', 'false']:
-            raise GPTInvalidResultException(content)
+            logger.error(f"JSONDecodeError: {content}")
```
Comment (jsntcy) [inline API/common/service_impl/knowledge.py]: I'm curious why it's Json decode error.

**Review by jsntcy — APPROVED** [review]

File: API/common/service_impl/knowledge.py (hunk 3 above)
Comment (wangzelin007) [reply API/common/service_impl/knowledge.py]: Add a new exception: `GPTInvalidBoolException`

**Pull request merged**